### DISCUSSION TITLE
All the results are 0 but workflow are normal

### DISCUSSION THREAD

**Discussion by mazidong1** [NHERI-SimCenter/SimCenterCommon#222] [R2D] [answered]
Dear developers，

After resolving the previous issues such as unifying acceleration units, I ran a regional simulation but ended up with all EDP, DV, etc. being 0. I checked the script log but couldn't find the problem. Is it because the PFA is too small? Is this normal? please help me have a check this time I really appreciate it！
![QQ图片20240911045012](https://github.com/user-attachments/assets/46820ea4-784d-4a1b-885f-5f6385152541)

[program output.docx](https://github.com/user-attachments/files/16952753/program.output.docx)

**Comment by jinyan1214** [comment]
I think you should look at the results folders and check if the PFA values are indeed small. The EDP values are stored in a file named "response.csv". If the PFA values are indeed small, I suggest you look at the EVENT.json in one of the "workdir.x" and confirm the ground motion magnitude is correct. Pay attention to the units.

**Reply by mazidong1** [reply]
Dear Jinyan,

Thank you for providing me with a solution to my question. I have carefully reviewed the unit of seismic motion (cm/s^2) I downloaded from the station and found that it is consistent with the acceleration unit (percentiles per second squared) I entered into the HAZ module in R2D. This indicates that there should be no problem with the acceleration unit. Considering your reminder to be extra careful with the unit, I have also adjusted the unit of assert in the GI module to cm and changed the projected area of the building in my BIM (CSV) to cm^2.

After running R2D again, the problem occurred in the DL module and displayed "Could't find EDP file for Buildings 5" (as shown in the figure below). I checked the responsive.csv and found that the PFA (559.808cm/s^2) on the first floor of Building id5 is actually equal to the PGA I input for seismic motion. Comparing with Example 2, the PFA and PGA of Building id1 are 6.6847m/s^2 and 21.9m/s^2, respectively, which makes me feel uneasy. Moreover, the TH_FILE in Eventjson of Example 2 is 3.28, while in mine it is 1.0. What does this TH_FILE mean?Is this where my problem lies?

After experiencing these simulations, I found that my understanding of the underlying logic of R2D is still insufficient. May I ask Jinyan, do you have any recommended software learning videos? Thank you for your guidance!

Best,
Zidong
![Z~OKO CL_%{%WHOCWI@JAPW](https://github.com/user-attachments/assets/181044fd-9102-4b1b-87db-fc208f0b7c6a)

![YVT}0 YN~0~E E770HGRYU9](https://github.com/user-attachments/assets/f04a96b9-45ec-454d-9e0a-b887beab3949)
![E7MIWH92{SIRR)M`(P{~E$2](https://github.com/user-attachments/assets/7b220547-6bb8-4941-ab75-1844462ac156)
[EVENT.json](https://github.com/user-attachments/files/16997047/EVENT.json)
[inputRWHALE.json](https://github.com/user-attachments/files/16997051/inputRWHALE.json)
[response.csv](https://github.com/user-attachments/files/16997053/response.csv)

**Reply by jinyan1214** [reply, marked as answer]
TH_file stands for time history file. Some resource to understand R2D better is the [developer manual](https://nheri-simcenter.github.io/R2D-Documentation/common/developer_manual/how_to_extend.html) in the R2D documentation, although some content in the documentation may be outdated. To figure out an answer to your questions, I suggested [running the R2D backend workflow driver manually](https://nheri-simcenter.github.io/R2D-Documentation/common/developer_manual/architecture/desktop/run-manually/runLocal.html) and confirming that the analysis in each step was performed as you expected. Please note that the command in "Run Locally on Your Desktop" is outdated and an example of the correct command is here:

`python $PWD/SimCenterBackendApplications/applications/Workflow/rWHALE.py ./R2DExamples/E1BasicHAZUS/input.json --registry $PWD/SimCenterBackendApplications/applications/Workflow/WorkflowApplications.json --referenceDir $PWD/input_data -w $PWD/tmp.SimCenter/Results`

I could not figure out the possible error according to the files attached to your post. If you would like to, please send me an email, and we can arrange a Zoom meeting so I can better assist you in solving the issue. You can find my email by right-clicking my name posted [here](https://simcenter.designsafe-ci.org/about/people/). 

Best,

Jinyan

**Reply by mazidong1** [reply]
Dear Jinyan,

Thank you very much for providing me with a timely solution.

I think I'll run it locally first (following the steps in the developer manual to install all the required software environments, and then manually operate), which may take me 2-3 days of effort, but it's worth it! If I still can't solve it, I think I really need to get in touch with you.

Anyway, thank you for your guidance. I hope you have a pleasant life.

Best,

Zidong

**Comment by mazidong1** [comment]
I came back to give feedback on Jinyan's strong help to me, because under his advice, the final operation was successful. I am really grateful.

Due to the use of real earthquake motion, the simulated area has not provided a valid result: it is displayed as 0.

Jinyan analyzed that the reason for this is not only that the seismic motion beneath the building may not be effective, but also that there are issues with the acceleration units displayed in the operation log, and that some structural parameters in the Hazus specification are not compatible with my building parameters.

Following his advice, I replaced the building and acceleration units with the same ones as in Example 2, and the final display showed successful operation.

I sincerely thank the Simcenter team and express my sincere gratitude to Jinyan!
![b27592810726ff437fdc5f68df8c928](https://github.com/user-attachments/assets/7501fbd6-c820-4622-a1ac-2e601512b93b)
